### ISSUE TITLE
Netrocks scp/sftp kex error : no match for method mac algo client -> server

### ISSUE THREAD

**Issue by chpasha** [elfmz/far2l#1805]
I'm getting this error since upgrading to Debian 12 and far2l 2.5.2 while connecting via ssh to old server running debian wheezy
![image](https://github.com/elfmz/far2l/assets/9569489/fb357b9f-6731-4bf1-9451-9e7f4a938091)
native terminal has no problem connecting, midnight commander has no problem as well. If I enable different legacy stuff, like suggested on the internet in /etc/ssh/ssh_config
```
KexAlgorithms diffie-hellman-group1-sha1,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1
Ciphers chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
 HostKeyAlgorithms ssh-rsa,ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,rsa-sha2-512,rsa-sha2-256
MACs hmac-md5,hmac-sha1,umac-64@openssh.com
```
 then I get ssh-agent authentication failed

**Comment by elfmz** [comment]
Try this: https://github.com/elfmz/far2l/discussions/1364#discussioncomment-4741879

**Comment by chpasha** [comment]
Hi, thank you for the answer. If you look at my code in description, I've already tried that (and not only). The issue under your link is a different one - about host key algo. I have it too but it is solved with entry in config or via option ssh-rsa supplied directly in advanced netrocks connection settings. The one from my screenshot comes after the first one is fixed and is about mac algorithms, and could be solved with cofiguration entry `MACs hmac-md5,hmac-sha1,umac-64@openssh.com` but it is not, because (this is my assumption) those algos are not present in libssh anymore (but are present in openssh-client or whatever lib the native clients and Midnight commander use)

**Comment by chpasha** [comment]
> maybe shell protocol could be a better option for you?
Unfortunately same problem

**Comment by unxed** [comment]
does console ssh client work in your case?

**Comment by chpasha** [comment]
>does console ssh client work in your case?

definitely, as well as midnight commander (which is my fallback option for now, if I have to get some files from those old servers)

**Comment by chpasha** [comment]
I don't update very often and didn't know about that change. Yes, it does work for me. Unfortunately, I have to test sftp connections from time to time, but I can do that in command line. Being able to access the old servers on a regular basis is more important to me, thanks 👍

**Comment by chpasha** [comment]
I don't know - it doesn't really solve the actual problem, but rather workarounds it for cases where sftp not involved. If there is no chance to fix it, then yes, it could be closed, I guess
